cc1plus: error: unrecognized command line option "-Wno-ignored-qualifiers"

Installing eventmachine 1.0.8 with native extensions

Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

    /Users/danielcalixto/.rbenv/versions/2.2.2/bin/ruby -r ./siteconf20150902-42332-1kloes5.rb extconf.rb
checking for rb_trap_immediate in ruby.h,rubysig.h... no
checking for rb_thread_blocking_region()... no
checking for ruby/thread.h... yes
checking for rb_thread_call_without_gvl() in ruby/thread.h... yes
checking for inotify_init() in sys/inotify.h... no
checking for __NR_inotify_init in sys/syscall.h... no
checking for writev() in sys/uio.h... yes
checking for rb_thread_fd_select()... yes
checking for rb_fdset_t in ruby/intern.h... yes
checking for pipe2() in unistd.h... no
checking for accept4() in sys/socket.h... no
checking for SOCK_CLOEXEC in sys/socket.h... no
checking for rb_wait_for_single_fd()... yes
checking for rb_enable_interrupt()... no
checking for rb_time_new()... yes
checking for sys/event.h... yes
checking for sys/queue.h... yes
CFLAGS= -O3 -Wno-error=shorten-64-to-32  -pipe  -Wall -Wextra -Wno-deprecated-declarations -Wno-ignored-qualifiers -Wno-unused-result
CPPFLAGS=-I/Users/danielcalixto/.rbenv/versions/2.2.2/include  -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE -D_DARWIN_UNLIMITED_SELECT -D_REENTRANT $(DEFS) $(cppflags) -Wall -Wextra -Wno-deprecated-declarations -Wno-ignored-qualifiers -Wno-unused-result
checking for clock_gettime()... no
checking for gethrtime()... no
creating Makefile

make "DESTDIR=" clean

make "DESTDIR="
compiling binder.cpp
cc1plus: error: unrecognized command line option "-Wno-ignored-qualifiers"
cc1plus: error: unrecognized command line option "-Wno-unused-result"
make: *** [binder.o] Error 1

make failed, exit code 2

Gem files will remain installed in /Users/danielcalixto/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/eventmachine-1.0.8 for inspection.
Results logged to /Users/danielcalixto/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/extensions/x86_64-darwin-11/2.2.0-static/eventmachine-1.0.8/gem_make.out
An error occurred while installing eventmachine (1.0.8), and Bundler cannot continue.
Make sure that `gem install eventmachine -v '1.0.8'` succeeds before bundling.

--- a/ext/extconf.rb
+++ b/ext/extconf.rb
@@ -181,6 +181,18 @@ else
   CONFIG['LDSHARED'] = "$(CXX) -shared"
 end
 
+# Platform-specific time functions
+if have_func('clock_gettime')
+  # clock_gettime is POSIX, but the monotonic clocks are not
+  have_const('CLOCK_MONOTONIC_RAW', 'time.h') # Linux
+  have_const('CLOCK_MONOTONIC', 'time.h') # Linux, Solaris, BSDs
+else
+  have_func('gethrtime') # Older Solaris and HP-UX
+end
+
+# Hack so that try_link will test with a C++ compiler instead of a C compiler
+TRY_LINK.sub!('$(CC)', '$(CXX)')
+
 # This is our wishlist. We use whichever flags work on the host.
 # In the future, add -Werror to make sure all warnings are resolved.
 # deprecated-declarations are used in OS X OpenSSL
@@ -197,23 +209,11 @@ end
 ).select do |flag|
   try_link('int main() {return 0;}', flag)
 end.each do |flag|
-  $CFLAGS << ' ' << flag
-  $CPPFLAGS << ' ' << flag
+  CONFIG['CXXFLAGS'] << ' ' << flag
 end
-puts "CFLAGS=#{$CFLAGS}"
-puts "CPPFLAGS=#{$CPPFLAGS}"
+puts "CXXFLAGS=#{CONFIG['CXXFLAGS']}"
 
-# Platform-specific time functions
-if have_func('clock_gettime')
-  # clock_gettime is POSIX, but the monotonic clocks are not
-  have_const('CLOCK_MONOTONIC_RAW', 'time.h') # Linux
-  have_const('CLOCK_MONOTONIC', 'time.h') # Linux, Solaris, BSDs
-else
-  have_func('gethrtime') # Older Solaris and HP-UX
-end
-
-# solaris c++ compiler doesn't have make_pair()
-TRY_LINK.sub!('$(CC)', '$(CXX)')
+# Solaris C++ compiler doesn't have make_pair()
 add_define 'HAVE_MAKE_PAIR' if try_link(<<SRC, '-lstdc++')
   #include <utility>
   using namespace std;

